### PR TITLE
refactor: consolidating our ts and tsup configs

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

Consolidates our many TS and `tsup` configs across these new packages.

I've also added a root `attw` script that we can run to run [attw](https://arethetypeswrong.github.io/) across every package.

### CODE REVIEW

**Comment by kanadgupta** [comment]
`attw` isn't running in our linting right? I don't see it in our dev-deps or anything!

**Comment by erunion** [comment]
Yeah `attw` isn't running anywhere, just setting it up for us to run manually on our machines right now. Want to build it into the linting step at some point but the Node10 resolution stuff throws a hard exit on the process so it'll be tough to build a status check around.

File: tsconfig.base.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "declaration": true,
+    "esModuleInterop": true,
+    "lib": ["ES2023"],
+    "module": "ESNext",
+    "moduleResolution": "Bundler",
+    "noImplicitAny": true,
+    "target": "ES2020"
```
Comment (kanadgupta) [inline tsconfig.base.json:10]: someday 🥺

```suggestion
    "strict": true,
    "target": "ES2020"
```

**Comment by kanadgupta** [comment]
> but the Node10 resolution stuff throws a hard exit on the process so it'll be tough to build a status check around

yeah I've noticed that and have been thinking of creating a feature request in the ATTW repo

File: packages/oas/tsconfig.json [hunk 2]
```diff
@@ -1,15 +1,9 @@
 {
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "allowJs": true,
     "baseUrl": "./src/",
-    "declaration": true,
-    "esModuleInterop": true,
     "lib": ["ES2023"],
```
Comment (kanadgupta) [inline packages/oas/tsconfig.json:5]: looks like we already have this in the base config:

```suggestion
```

File: packages/oas-extensions/tsconfig.json [hunk 3]
```diff
@@ -1,14 +1,8 @@
 {
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "allowJs": true,
     "baseUrl": "./src/",
```
Comment (kanadgupta) [inline packages/oas-extensions/tsconfig.json:4]: it looks like all of these have the same `baseUrl`, should we move that to the base config?

File: packages/oas-normalize/tsconfig.json [hunk 4]
```diff
@@ -1,17 +1,12 @@
 {
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "allowJs": true,
     "baseUrl": "./src",
-    "declaration": true,
-    "esModuleInterop": true,
-    "module": "ESNext",
-    "moduleResolution": "bundler",
+    "lib": ["DOM"],
```
Comment (kanadgupta) [inline packages/oas-normalize/tsconfig.json:5]: could we stick with the base `lib` value? why does this library need DOM APIs?

```suggestion
```

File: packages/oas-to-har/tsconfig.json [hunk 5]
```diff
@@ -1,17 +1,10 @@
 {
+  "extends": "../../tsconfig.base.json",
   "compilerOptions": {
-    "allowJs": true,
     "baseUrl": "./src",
-    "declaration": true,
     "downlevelIteration": true,
-    "esModuleInterop": true,
     "lib": ["DOM", "ES2020"],
```
Comment (kanadgupta) [inline packages/oas-to-har/tsconfig.json:6]: could we stick with the base `lib` value? why does this library need DOM APIs?

```suggestion
```

File: tsconfig.base.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
```
Comment (kanadgupta) [inline tsconfig.base.json:3]: also your call on this but I've been using this as a handy reference lately: https://www.totaltypescript.com/tsconfig-cheat-sheet#building-for-a-library-in-a-monorepo

and it recommends adding the following options in monorepos:

```suggestion
    "allowJs": true,
    "composite": true,
    "sourceMap": true,
    "declarationMap": true
```

File: packages/oas-normalize/tsconfig.json (hunk 4 above)
Comment (erunion) [reply packages/oas-normalize/tsconfig.json:5]: `oas-normalize` uses `fetch` which only exists in `DOM`.

File: packages/oas-extensions/tsconfig.json (hunk 3 above)
Comment (erunion) [reply packages/oas-extensions/tsconfig.json]: Moving these to the base config makes TS think that our base URL for the source is `./src` not `packages/*/src`.

Comment (kanadgupta) [reply packages/oas-extensions/tsconfig.json]: huh, I got it working here: https://github.com/readmeio/api/pull/733/files#diff-0b280a445be167f54cd916c0718c952b8e0d4ca9621903d05eec25efd4c6ed6d

**Pull request merged**